### PR TITLE
Add verbose parameter to lr_scheduler type hints

### PR DESCRIPTION
**Pull request by kbrose**
Re-open of https://github.com/pytorch/pytorch/pull/60046 which stalled out, gained merge conflicts, and I had since deleted my fork of pytorch.

In the current version of pytorch, you get a type error when using `lr_scheduler.<some scheduler>(..., verbose=True)` even though this is allowed by the code. This MR results in no type error, as expected.

### CODE REVIEW

**Comment by saketh-are** [comment]
@iramazanli Would you be able to take a look at this?

**Comment by kbrose** [comment]
friendly ping
